b2share/modules/deposit/api.py

```diff
@@ -74,6 +81,9 @@ def build_deposit_schema(self, record):
     @classmethod
     def create(cls, data, id_=None):
         """Create a deposit."""
+        # check that the status field is not set
+        assert 'publication_state' not in data
```

Comment:
But yes we can also raise an exception instead. The would should be similar, i.e fail on bad input.